core/client/src/main/java/alluxio/client/block/RetryHandlingBlockWorkerClient.java

```diff
@@ -249,15 +245,15 @@ public Void call(BlockWorkerClientService.Client client)
   }
 
   @Override
-  public String requestBlockLocation(final long blockId, final long initialBytes)
-      throws IOException {
+  public String requestBlockLocation(final long blockId, final long initialBytes,
+      final int writeTier) throws IOException {
```

Comment:
The challenge with specifying a tier on remote writes is  the netty protocol isn't versioned right now, so we wouldn't be backward compatible with older clients.